### PR TITLE
Add address field to capabilities

### LINKED ISSUE #721: Getting the `Address` from a `Capability`
Currently there's no way to get the `Address` from a `Capability`.

My use case was that I had a function which accepts a `Capability`, and I wanted to emit an event that records the address, like this:

```
pub registerCapability(c: Capability) {
    emit CapabilityRegistered(address: c.getAddress()) // this isn't possible
}
```

### PR DESCRIPTION
**Pull request by ceelo777**
Closes #721

## Description
This PR should add the ability to grab the address field from a capability.
______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Comment by ceelo777** [comment]
So this isn't finished @turbolent - I'm having a weird issue with the test I wrote -

`func TestCheckCapability_address(t *testing.T) {

	t.Parallel()
	t.Run("check address", func(t *testing.T) {
		checker, err := ParseAndCheckWithPanic(t,
			`		  			
				let capability: Capability = panic("")
				let addr = capability.address
			`,
		)
		require.NoError(t, err)

		addrType := RequireGlobalType(t, checker.Elaboration, "addr")
		require.Equal(t, sema.AddressType{}, addrType)
	})
}
`

I'm getting this error - "global type 'addr' missing" even though it seems that I'm initializing it? Not sure exactly why this is happening.

File: runtime/tests/checker/capability_test.go [hunk 1]
```diff
@@ -434,3 +434,20 @@ func TestCheckCapability_check(t *testing.T) {
...
+	t.Run("check address", func(t *testing.T) {
+		checker, err := ParseAndCheckWithPanic(t,
+			`		  			
+				let capability: Capability = panic("")
+				let addr = capability.address
+			`,
+		)
+		require.NoError(t, err)
+
+		addrType := RequireGlobalType(t, checker.Elaboration, "addr")
```
Comment (SupunS) [inline runtime/tests/checker/capability_test.go:450]: I think the issue is, `addr` is a variable but not a type. Using `RequireGlobalValue` would solve the test failure.  
```suggestion
		addrType := RequireGlobalValue(t, checker.Elaboration, "addr")
```

File: runtime/tests/checker/capability_test.go [hunk 2]
```diff
@@ -434,3 +434,20 @@ func TestCheckCapability_check(t *testing.T) {
...
+		require.Equal(t, sema.AddressType{}, addrType)
```
Comment (turbolent) [inline runtime/tests/checker/capability_test.go:451]:
```suggestion
		require.Equal(t, &sema.AddressType{}, addrType)
```

**Review by turbolent — COMMENTED** [review]
Thank you for contributing this!

Looks good, just the test needs to be fixed and then it can be merged

File: runtime/tests/checker/capability_test.go (hunk 1 above)
Comment (turbolent) [reply runtime/tests/checker/capability_test.go:450]: @ceelo777 like Supun pointed out here, you want to get the type of the variable declaration, which is a value. 

`RequireGlobalType` is used to the the type introduced by a type declaration, e.g. a struct or resource declaration.

File: runtime/tests/checker/capability_test.go (hunk 2 above)
Comment (ceelo777) [reply runtime/tests/checker/capability_test.go:451]: Thanks! It worked, updating the PR now.

**Comment by turbolent** [comment]
@ceelo777 Looks good, glad the  test passes now. Just one last thing: Could you please add a test for the interpreter?

**Comment by ceelo777** [comment]
To be honest, I'm not sure if I did this interpreter test correctly but it passed - anything else I should add to it?

File: runtime/tests/interpreter/capability_test.go [hunk 3]
```diff
@@ -657,3 +657,46 @@ func TestInterpretCapability_check(t *testing.T) {
...
+
+				fun nonExistent() : Address {
+					return account.getCapability(/public/nonExistent).address
+				}				
+			`,
+		)
+
+		t.Run("single", func(t *testing.T) {
+			_, err := inter.Invoke("single")
+			require.NoError(t, err)
```
Comment (turbolent) [inline runtime/tests/interpreter/capability_test.go:689]: Here and in the tests below, assert that the result value, that is currently being ignored, is in fact of type `interpreter.AddressValue`:

```suggestion
			value, err := inter.Invoke("single")
			require.NoError(t, err)
			
			require.IsType(t, interpreter.AddressValue{}, value)
```

**Review by turbolent — COMMENTED** [review]
The test looks good! 👍 

Good idea to use the `testAccount` helper.
One last thing would be to check the result values, then we can merge

File: runtime/tests/interpreter/capability_test.go (hunk 3 above)
Comment (turbolent) [reply runtime/tests/interpreter/capability_test.go]: Resolved in 11655c8

File: runtime/interpreter/value.go [hunk 4]
```diff
@@ -7069,7 +7069,8 @@ func (v CapabilityValue) GetMember(inter *Interpreter, _ func() LocationRange, n
 			borrowType = inter.ConvertStaticToSemaType(v.BorrowType).(*sema.ReferenceType)
 		}
 		return inter.capabilityCheckFunction(v.Address, v.Path, borrowType)
-
+	case "address":
```
Comment (turbolent) [inline runtime/interpreter/value.go:7072]:
```suggestion

	case "address":
```

Comment (turbolent) [inline runtime/interpreter/value.go:7073]:
```suggestion

	case "address":
```

File: runtime/sema/type.go [hunk 5]
```diff
@@ -6527,6 +6527,10 @@ const capabilityTypeCheckFunctionDocString = `
 Returns true if the capability currently targets an object that satisfies the given type, i.e. could be borrowed using the given type
 `
 
+const addressTypeCheckFunctionDocString = `
+Returns the address for a capability
```
Comment (turbolent) [inline runtime/sema/type.go:6531]:
```suggestion
The address of the capability
```

**Review by turbolent — APPROVED** [review]
Nice work, thanks again for adding this @ceelo777!

**Pull request merged**